flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java

```diff
@@ -75,32 +95,436 @@ public void clean() {
+    String explainExpected = "FilterPushDown,the filters :ref(name=\"id\") != 1]]], fields=[id, data])";
+    assertTrue("explain should contains FilterPushDown", explainNE.contains(explainExpected));
+
+    List<Object[]> resultNE = sql(sqlNE);
+    assertEquals("should have 2 record", 2, resultNE.size());
+
+    List<Object[]> expectedNE = Lists.newArrayList();
+    expectedNE.add(new Object[] {2, "b"});
+    expectedNE.add(new Object[] {3, null});
+    assertArrayEquals("Should produce the expected record", resultNE.toArray(), expectedNE.toArray());
+    assertEquals("Should create only one scan", 1, scanEventCount);
+    assertEquals("Should push down expected filter", "id != 1", ExpressionsUtil.describe(lastScanEvent.filter()));
+
+    String sqlNotEqualNull = String.format("SELECT * FROM %s WHERE data <> NULL ", TABLE_NAME);
+    String explainNotEqualNull = getTableEnv().explainSql(sqlNotEqualNull);
+    assertFalse("explain should not contains FilterPushDown", explainNotEqualNull.contains(
+        expectedFilterPushDownExplain));
+  }
+
+  @Test
+  public void testFilterPushDownAnd() {
+    String sqlAnd = String.format("SELECT * FROM %s WHERE id = 1 AND data = 'a' ", TABLE_NAME);
+    String explainAnd = getTableEnv().explainSql(sqlAnd);
+    String explainExpected =
+        "FilterPushDown,the filters :ref(name=\"id\") == 1,ref(name=\"data\") == \"a\"]]], fields=[id, data])";
+    assertTrue("explain should contains FilterPushDown", explainAnd.contains(explainExpected));
+
+    List<Object[]> resultAnd = sql(sqlAnd);
+    assertEquals("should have 1 record", 1, resultAnd.size());
+    assertArrayEquals("Should produce the expected record", resultAnd.get(0), new Object[] {1, "a"});
+
+    assertEquals("Should create only one scan", 1, scanEventCount);
+    assertEquals("Should push down expected filter", "(id = 1 AND data = 'a')",
+        ExpressionsUtil.describe(lastScanEvent.filter()));
+  }
+
+  @Test
+  public void testFilterPushDownOr() {
+    String sqlOr = String.format("SELECT * FROM %s WHERE id = 1 OR data = 'b' ", TABLE_NAME);
+    String explainOr = getTableEnv().explainSql(sqlOr);
+    String explainExpected =
+        "FilterPushDown,the filters :(ref(name=\"id\") == 1 or ref(name=\"data\") == \"b\")]]], fields=[id, data])";
+    assertTrue("explain should contains FilterPushDown", explainOr.contains(explainExpected));
+
+    List<Object[]> resultOr = sql(sqlOr);
+    assertEquals("should have 2 record", 2, resultOr.size());
+
+    List<Object[]> expectedOR = Lists.newArrayList();
+    expectedOR.add(new Object[] {1, "a"});
+    expectedOR.add(new Object[] {2, "b"});
+    assertArrayEquals("Should produce the expected record", resultOr.toArray(), expectedOR.toArray());
+
+    assertEquals("Should create only one scan", 1, scanEventCount);
+    assertEquals("Should push down expected filter", "(id = 1 OR data = 'b')",
+        ExpressionsUtil.describe(lastScanEvent.filter()));
```

Comment:
It looks like `describe` was moved just for tests. I don't think that was needed. If you need to produce a string from an expression, why not just use `toString`? That's what other assertions use, like the one for `explainExpected` above.

Can you try reverting the `describe` change?